### ISSUE TITLE
Fix S112 FN when using null-coalesce operator

### ISSUE THREAD

**Issue by nalka0** [SonarSource/sonar-dotnet#7910]
### Description

S112 isn't raised when `throw new Exception();` is used with null-coalesce operator

### Repro steps

```cs
var test = "" ?? throw new Exception(); // Doesn't raise S112
throw new Exception(); // Raises S112
```

### Expected behavior

S112 is raised on both lines

### Actual behavior

S112 is raised only on the second line

### Known workarounds

Not using null-coalesce operator

### Related information

* SonarLint for Visual Studio 2022 Version 7.2.0.76209
* Visual Studio Professional 2022 Version 17.6.4
* .Net Framework 4.7.2 / .Net 6.0
* Windows 11

**Comment by pavel-mikula-sonarsource** [comment]
Hi @nalka0,

Thank you for reporting this case, I confirm it as FN.

**Comment by pavel-mikula-sonarsource** [comment]
Other scenarios:
Lambda
```
_ => throw new Exception()
```

and arrow methods
```
public void Go() => throw new Exception();
```
